### ISSUE TITLE
Has anyone used this with a TypeScript project such as Angular?

### ISSUE THREAD

**Issue by kamok** [rosiejs/rosie#71]
Any challenges or gotchas? I couldn't find anything as clean as rosie for .ts

**Comment by kamok** [comment]
So I've tackle this problem a bit, trying to use it with my NativeScript project. I've followed the instructions here: https://stackoverflow.com/questions/34499382/importing-javascript-library-into-nativescript

Requiring this in my factory definition:
```
const FactoryModule = require('rosie');
const Factory = new FactoryModule.Factory();

export default Factory.define('employer').sequence('id').attrs({
  name: 'Employer Name',
});
```

But when I run my specs, I still get `Error: Could not find module 'rosie'`

**Comment by kamok** [comment]
I was able to solve this problem. Here's how to load rosie into a NativeScript project powered by TypeScript.

1. `npm i rosie --save` - not save dev!
2. `npm i @types/rosie --save` - not save dev!
3.  In factory, require like this:
```
const FactoryModule = require('rosie');
const Factory = new FactoryModule.Factory();

export default Factory.sequence('id').attrs({
   name: 'Foo'
})
```
4. In spec file, use like this:
```
import MyFactory from '~/tests/factories/my.factory';
MyFactory.build({ name: 'Bar' })
```

MyFactory can be anything and it should still work, no need to declare inside your factory file.

5. You might need to add the following to your tsconfig
`"noImplicitAny": false,`

That's it. Closing.
